### DISCUSSION TITLE
The console output is garbled

### DISCUSSION THREAD

**Discussion by Jiang-Nian-Hua** [quarkusio/quarkus#43794] [Community]
<img width="971" alt="image" src="https://github.com/user-attachments/assets/cdbc5ec4-73d8-4712-ab68-655f0a06677f">

It works fine in local testing, but this issue occurs when using binary packaging (GraalVM compilation).

**Comment by cescoffier** [comment]
Looks like an encoding issue.

WDTY @dmlloyd ?

**Reply by dmlloyd** [reply]
Yes I agree this is an encoding problem specific to native image. It may be excluding the desired charset and falling back to `US-ASCII` for example.

**Reply by Jiang-Nian-Hua** [reply]
When building, I added `quarkus.native.additional-build-args=--add-charset=UTF-8` to my configuration file, but it still doesn't work.

**Reply by Jiang-Nian-Hua** [reply]
<img width="1074" alt="image" src="https://github.com/user-attachments/assets/1c8df43d-8f0c-4b65-846d-10a28b8637b6">

**Reply by Jiang-Nian-Hua** [reply]
```
FROM ************.dkr.ecr.ap-northeast-1.amazonaws.com/scash-build/quarkus-builder:latest AS TEMP_BUILD_IMAGE

WORKDIR /project
COPY . .

ENV GRADLE_OPTS="-Xmx4g -Dorg.gradle.daemon=false"

RUN gradle build --no-daemon -Dquarkus.package.type=native -Dquarkus.container-image.build=false -Dquarkus.container-image.push=false -Dquarkus.native.add-all-charsets=true -Dquarkus.native.native-image-xmx=4g


FROM **********.dkr.ecr.ap-northeast-1.amazonaws.com/scash-poc/ubuntu:22.04
WORKDIR /work/

RUN chown 1001 /work \
    && chmod "g+rwX" /work \
    && chown 1001:root /work

COPY --from=TEMP_BUILD_IMAGE --chown=1001:root /project/build/*-runner /work/application

EXPOSE 8080
USER 1001

ENTRYPOINT ["./application", "-Dquarkus.http.host=0.0.0.0","-Dquarkus.log.console.charset=UTF-8"]
```

**Comment by dmlloyd** [comment]
Recent versions of Java will always default to `UTF-8` (see [JEP 400](https://openjdk.org/jeps/400)) but this is not true of older versions, and may not be completely true of GraalVM. Try setting `-Dfile.encoding=UTF-8` when you run the application. This is the ideal solution, because it means that all output (not just logging) would be using the correct output encoding.

If that doesn't work, you might try setting the system environment's charset using the `LANG` environment variable (assuming Linux or MacOS). An example would be `LANG=en_US.UTF-8` (or whatever system language you want to use - the `UTF-8` is the important part).

**Reply by Jiang-Nian-Hua** [reply]
OK THX~
